tests/utils/git_test.py

```diff
@@ -140,41 +154,66 @@ def test_fetch_with_error(self):
 
     def test_checkout_successful(self, setup_teardown_local_branch):
         git.checkout(TestGit.tmp_src_dir, branch=TestGit.local_branch)
+        assert git.current_branch(TestGit.tmp_src_dir) == TestGit.local_branch
 
     def test_checkout_with_error(self):
         branch = "this-branch-doesnt-actually-exist"
         with pytest.raises(exceptions.SupplyError) as exc:
             git.checkout(TestGit.tmp_src_dir, branch=branch)
         assert exc.value.args[0] == f"Could not checkout [{branch}]. Do you have uncommitted changes?"
 
-    def test_rebase(self, setup_teardown_rebase):
-        git.rebase(TestGit.tmp_src_dir, remote="origin", branch="master")
-
-    def test_pull(self, setup_teardown_rebase):
-        git.pull(TestGit.tmp_src_dir, remote="origin", branch="master")
-
-    def test_pull_ts(self, setup_teardown_rebase):
-        git.pull_ts(TestGit.tmp_src_dir, "20160101T110000Z", remote="origin", branch="master")
-
-    def test_checkout_revision(self, setup_teardown_rebase):
+    def test_checkout_revision(self):
         git.checkout_revision(TestGit.tmp_src_dir, revision="bd368741951c643f9eb1958072c316e493c15b96")
+        assert "bd368741951c643f9eb1958072c316e493c15b96" == git.head_revision(TestGit.tmp_src_dir)
 
-    def test_head_revision(self, setup_teardown_head_revision):
+    def test_checkout_branch(self):
+        git.checkout_branch(TestGit.tmp_src_dir, remote="upstream", branch=TestGit.remote_branch)
+        assert "be3eeda86e12616fa744f6cde7fcfe41b3a1c963" == git.head_revision(TestGit.tmp_src_dir)
+
+    def test_head_revision(self):
         # Apple Git 'core.abbrev' defaults to return 7 char prefixes (09980cd)
         # Linux defaults to return 8 char prefixes (09980cd5)
+        git.checkout(TestGit.tmp_src_dir, branch="2.6.0")
         assert git.head_revision(TestGit.tmp_src_dir).startswith("09980cd")
 
-    def test_list_remote_branches(self):
-        assert TestGit.remote_branch in git.branches(TestGit.tmp_src_dir, remote=True)
-
-    def test_list_local_branches(self, setup_teardown_local_branch):
-        assert TestGit.local_branch in git.branches(TestGit.tmp_src_dir, remote=False)
+    def test_pull_ts(self):
+        git.pull_ts(TestGit.tmp_src_dir, "20160101T110000Z", remote="origin", branch="master")
+        assert "28474f4f097106ff3507be35958db0c3c8be0fc6" == git.head_revision(TestGit.tmp_src_dir)
 
-    def test_list_tags_with_tags_present(self):
-        expected_tags = ["2.6.0", "2.5.0", "2.4.0", "2.3.1", "2.3.0"]
-        tags = git.tags(TestGit.tmp_src_dir)
-        for tag in expected_tags:
-            assert tag in tags
+    @mock.patch("esrally.utils.process.run_subprocess_with_logging")
```

Comment:
I can't think of a way to properly test rebase or pull without doing things like pushing commits to a remote from _another_ cloned dir, which seems like it's getting a bit messy for unit test territory.